scheme/sevsnp/evidence_handler.go

```diff
@@ -0,0 +1,530 @@
+	// RawValue comparison
+	if refM.Val.RawValue != nil {
+		if evM.Val.RawValue == nil {
+			return false
+		}
+
+		refDigest, _ := refM.Val.RawValue.GetBytes()
+		return evM.Val.RawValue.CompareAgainstReference(refDigest, nil)
+	}
+
+	// Digests comparison
+	if refM.Val.Digests != nil {
+		if evM.Val.Digests == nil {
+			return false
+		}
+
+		return evM.Val.Digests.CompareAgainstReference(*refM.Val.Digests)
+	}
+
+	// SVN comparison
+	if refM.Val.SVN != nil {
+		if evM.Val.SVN == nil {
+			log.Debugf("evidence doesn't have SVN")
+			return false
+		}
+
+		if c, ok := evM.Val.SVN.Value.(*comid.TaggedSVN); ok {
+			if r, ok := refM.Val.SVN.Value.(*comid.TaggedSVN); ok {
+				return c.CompareAgainstRefSVN(*r)
+			} else if r, ok := refM.Val.SVN.Value.(*comid.TaggedMinSVN); ok {
+				return c.CompareAgainstRefMinSVN(*r)
+			} else {
+				log.Debugf("unknown refVal SVN type")
+				return false
+			}
+		} else if c, ok := evM.Val.SVN.Value.(*comid.TaggedMinSVN); ok {
+			if r, ok := refM.Val.SVN.Value.(*comid.TaggedMinSVN); ok {
+				return c.Equal(*r)
+			}
+			log.Debugf("can't compare TaggedMinSVN against TaggedSVN")
+			return false
+		} else {
+			log.Debugf("unknown evidence SVN type")
+			return false
+		}
+	}
+
+	return true
+}
+
+func compareTcb(refM comid.Measurement, evM comid.Measurement) bool {
+	if refM.Val.SVN == nil {
+		log.Errorf("reference doesn't have SVN")
+		return false
+	}
+
+	if evM.Val.SVN == nil {
+		log.Errorf("evidence doesn't have SVN")
+		return false
+	}
+
+	refTcbParts, err := transformSVNtoTCB(*refM.Val.SVN)
+	if err != nil {
+		log.Errorf("could not transform reference SVN to TCB parts: %v", err)
+		return false
+	}
+
+	evTcbParts, err := transformSVNtoTCB(*evM.Val.SVN)
+	if err != nil {
+		log.Errorf("could not transform evidence SVN to TCB parts: %v", err)
+	}
+
+	if evTcbParts.BlSpl < refTcbParts.BlSpl ||
+		evTcbParts.SnpSpl < refTcbParts.SnpSpl ||
+		evTcbParts.TeeSpl < refTcbParts.TeeSpl ||
+		evTcbParts.UcodeSpl < refTcbParts.UcodeSpl {
+		return false
+	}
+
+	return true
+}
+
+// AppraiseEvidence confirms if the claims in the evidence match with the provisioned
+// reference values.
+//
+// Appraisal can confirm if the evidence is genuinely generated by AMD
+// hardware and if SEV-SNP enables memory encryption. As such, set the
+// "Hardware" and "RuntimeOpaque" values in the trustworthiness vector;
+// we can't infer other aspects of the vector from SEV-SNP evidence alone.
+func (o EvidenceHandler) AppraiseEvidence(
+	ec *proto.EvidenceContext,
+	endorsementsStrings []string,
+) (*ear.AttestationResult, error) {
+	var (
+		err         error
+		evidenceMap map[string]interface{}
+	)
+
+	refVal, err := refvalToComidTriple(endorsementsStrings)
+	if err != nil {
+		return nil, err
+	}
+
+	evidence, err := evidenceToComidTriple(ec)
+	if err != nil {
+		return nil, err
+	}
+
+	result := handler.CreateAttestationResult(SchemeName)
+
+	appraisal := result.Submods[SchemeName]
+
```

Comment:
Just a question: Is this init to default values or more than this..?
Can you please specify, perhaps a short comment might help as well!

Reply:
@yogeshbdeshpande Thanks for your feedback, but I wasn't sure how to address it.

This one isn't initializing. The initialization of the Trustworthiness Vector happens below. Could you kindly provide more details on what I should do here? Thanks!